src/plugin/nerd_plugin.js

```
@@ -254,7 +254,7 @@ class NerdPluginVirtual extends EventEmitter {
 
     const valid = yield this.balance.checkAndSettleOutgoing(transfer.amount)
     const validAmount = (typeof transfer.amount === 'string' &&
-      !isNaN(transfer.amount - 0) && (transfer.amount - 0 >= 0))
+      !isNaN(transfer.amount - 0) && (transfer.amount - 0 > 0))
```

Is `amount - 0` used to cast to Number? If you're after brevity `+amount` would work, too. That said, I prefer `Number(amount)` for readability.

However, since this is user input, we should be a bit more paranoid, note:

``` js
> Number('0x10')
16
> parseFloat('0x10')
0
> JSON.parse('0x10')
SyntaxError: Unexpected token x in JSON at position 1
```

``` js
> Number('')
0
> parseFloat('')
NaN
> JSON.parse('')
SyntaxError: Unexpected end of JSON input
```

It's not good that `Number` and `parseFloat` quietly accept these bogus values as valid inputs. You could use `JSON.parse` and check that the return value is of type `"number"`. You'd want to catch and rethrow the error though to give a more useful message. Or you could test the value against a regex first, but note that we may want to allow scientific notation, e.g. `1e15`.

---

added a function in `util` that `JSON.parse`es a string and returns it only if the parse was successful. The functions where this is used already check to make sure that it isn't NaN, and if it is then they will emit an `InvalidFieldsError`.